src/zcash/address/unified.cpp

```diff
@@ -23,30 +41,26 @@ std::optional<std::pair<ZcashdUnifiedSpendingKey, HDKeyPath>> ZcashdUnifiedSpend
     auto saplingKey = SaplingExtendedSpendingKey::ForAccount(seed, bip44CoinType, accountId);
     usk.saplingKey = saplingKey.first;
 
-    return std::make_pair(usk, saplingKey.second);
+    return usk;
 }
 
-ZcashdUnifiedFullViewingKey ZcashdUnifiedSpendingKey::ToFullViewingKey() const {
-    ZcashdUnifiedFullViewingKey ufvk;
+UnifiedFullViewingKey ZcashdUnifiedSpendingKey::ToFullViewingKey() const {
+    UnifiedFullViewingKeyBuilder builder;
 
-    if (transparentKey.has_value()) {
-        auto extPubKey = transparentKey.value().Neuter();
+    auto extPubKey = transparentKey.Neuter();
+    builder.AddTransparentKey(CChainablePubKey::FromParts(extPubKey.chaincode, extPubKey.pubkey).value());
```

Comment:
Confirmed: `std::nullopt` means "pubkey is not compressed", which never happens for us because we always derive compressed keys. So ignoring it here and throwing if `.value()` is called on `std::nullopt` is fine.